client/jetpack-connect/authorize.js

```diff
@@ -256,11 +256,12 @@ export class JetpackAuthorize extends Component {
 	}
 
 	shouldAutoAuthorize() {
-		const { alreadyAuthorized, authApproved } = this.props.authQuery;
-
+		const { alreadyAuthorized, authApproved, from } = this.props.authQuery;
 		return (
 			this.isSso() ||
-			this.isWoo() ||
+			( 'woocommerce-services-auto-authorize' === from ||
+				( ! config.isEnabled( 'jetpack/connect/woocommerce' ) &&
+					'woocommerce-setup-wizard' === ' from' ) ) ||
```

Comment:
@justinshreve Looks like a bug slipped in here:

```js
'woocommerce-setup-wizard' === ' from'
// ☝️always evaluates to false
```

Reply:
@sirreal Thanks! Dan caught that earlier.